### PR TITLE
[SPARK-5769] Set params in constructors and in setParams in Python ML pipelines

### PR DESCRIPTION
**Pull request by mengxr**
This PR allow Python users to set params in constructors and in setParams, where we use decorator `keyword_only` to force keyword arguments. The trade-off is discussed in the design doc of SPARK-4586.

Generated doc:
screen shot 2015-02-12 at 3 06 58 am

CC: @davies @rxin

### CODE REVIEW

**Comment by mengxr** [comment]
@davies I updated the constructor doc signature. Auto-completion works in both iPython and IntelliJ now. The only issue is that the `help` doc in iPython still contains the doc signature. I think this is a minor one.

File: python/pyspark/ml/classification.py [hunk 1]
```diff
@@ -32,22 +32,46 @@ class LogisticRegression(JavaEstimator, HasFeaturesCol, HasLabelCol, HasPredicti
...
     >>> from pyspark.sql import Row
     >>> from pyspark.mllib.linalg import Vectors
-    >>> dataset = sqlCtx.inferSchema(sc.parallelize([ \
-            Row(label=1.0, features=Vectors.dense(1.0)), \
-            Row(label=0.0, features=Vectors.sparse(1, [], []))]))
-    >>> lr = LogisticRegression() \
-            .setMaxIter(5) \
-            .setRegParam(0.01)
-    >>> model = lr.fit(dataset)
+    >>> df = sqlCtx.inferSchema(sc.parallelize([
```
Comment (davies) [inline python/pyspark/ml/classification.py]: Wait for #4557, then you could user new API for convert RDD into DataFrame:
```
dataset = rdd.toDF()
```

File: python/pyspark/ml/feature.py [hunk 2]
```diff
@@ -29,18 +29,45 @@ class Tokenizer(JavaTransformer, HasInputCol, HasOutputCol):
...
+    >>> tokenizer.setParams("text")
+    Traceback (most recent call last):
+        ...
+    TypeError: Method setParams forces keyword arguments.
     """
 
     _java_class = "org.apache.spark.ml.feature.Tokenizer"
 
+    @keyword_only
+    def __init__(self, inputCol="input", outputCol="output"):
```
Comment (davies) [inline python/pyspark/ml/feature.py]: Is it possible to use 'input' instead of 'inputCol'? The same for `outputCol`

Comment (mengxr) [inline python/pyspark/ml/feature.py]: It may cause troubles in other parameters that specify column names, for example `featuresCol` and `labelCol`. This is not part of this PR.

Comment (davies) [inline python/pyspark/ml/feature.py]: Make sense.

File: python/pyspark/ml/classification.py (hunk 1 above)
Comment (davies) [inline python/pyspark/ml/classification.py]: #4557 is closed, so you change update the tests (examples) to use new API.

**Comment by davies** [comment]
LGTM， thanks！

**Comment by mengxr** [comment]
Merged into master and branch-1.3.
